apps/web/components/SideNavigationBar/index.tsx

```diff
@@ -96,27 +97,39 @@ export default function SideNav() {
     setIsCreateNewBudgetModalVisible(true);
   };
 
-  const successData =
-    data?.pages?.flatMap(
-      ({ items }: ItemType) =>
-        items &&
-        items.map(({ icon, name, id, isFavourite }) => ({
-          ComponentToRender: (
-            <>
-              <IconStyled icon={validate(icon) ? icon : "help"} iconSize={24} />
-              <SpanStyled>{name}</SpanStyled>
-              <Favourite
-                isFav={isFavourite}
-                budgetId={id.value}
-                activeHref={`/budgets/${id.value}`}
-              />
-            </>
-          ),
-          href: `/budgets/${id.value}`,
-          id: id.value,
-          ref: lastBudgetRef,
-        }))
-    ) ?? [];
+  const mapToBudgetsComponents = (budgets: BudgetType[]) => {
+    return budgets.map(({ icon, name, id, isFavourite }) => ({
+      ComponentToRender: (
+        <>
+          <IconStyled icon={validate(icon) ? icon : "help"} iconSize={24} />
+          <SpanStyled>{name}</SpanStyled>
+          <Favourite
+            isFav={isFavourite}
+            budgetId={id.value}
+            activeHref={`/budgets/${id.value}`}
+          />
+        </>
+      ),
+      href: `/budgets/${id.value}`,
+      id: id.value,
+      ref: lastBudgetRef,
+    }));
```

Comment:
This is a bug that's been sitting here for a while, might as well fix it now. This adds last budget ref to every single budget on the list, instead of the last one.
```suggestion
    return budgets.map(({ icon, name, id, isFavourite }, index, list) => ({
      ComponentToRender: (
        <>
          <IconStyled icon={validate(icon) ? icon : "help"} iconSize={24} />
          <SpanStyled>{name}</SpanStyled>
          <Favourite
            isFav={isFavourite}
            budgetId={id.value}
            activeHref={`/budgets/${id.value}`}
          />
        </>
      ),
      href: `/budgets/${id.value}`,
      id: id.value,
      ref: index === list.length - 1 ? lastBudgetRef : null,
    }));
```